src/mute/muteModel.js

```diff
@@ -18,27 +21,60 @@ export const getMute = (state: PerAccountState): MuteState => state.mute;
 // Getters.
 //
 
-export const isTopicMuted = (streamId: number, topic: string, mute: MuteState): boolean =>
-  mute.get(streamId)?.has(topic) ?? false;
+export function getTopicVisibilityPolicy(
+  mute: MuteState,
+  streamId: number,
+  topic: string,
+): UserTopicVisibilityPolicyT {
+  return mute.get(streamId)?.get(topic) ?? UserTopicVisibilityPolicy.None;
+}
+
+export function isTopicMuted(streamId: number, topic: string, mute: MuteState): boolean {
+  const policy = getTopicVisibilityPolicy(mute, streamId, topic);
+  switch (policy) {
+    // FlowIssue: sad that we end up having to write numeric literals here :-/
+    //   But the most important thing to get from the type-checker here is
+    //   that the ensureUnreachable works -- that ensures that when we add a
+    //   new possible value, we'll add a case for it here.  Couldn't find a
+    //   cleaner way to write this that still accomplished that.
```

Comment:
> Is this the same issue as [#5384 (comment)](https://github.com/zulip/zulip-mobile/pull/5384#discussion_r875147220) ?

Ah, yep. I knew I'd run into that issue recently, went looking and found that other commit https://github.com/zulip/zulip-mobile/commit/04c5a03d27a580182dce70c4b99260227614fb1d, but clearly didn't find the right search because this is what I was actually thinking of :slightly_smiling_face:

> Should we leave a note about trying Flow enums with the flowlint rule `require-explicit-enum-switch-cases`, new in Flow 153?

I think the main thing is just Flow enums themselves -- I believe that's already enough that Flow will point out an error if a switch isn't exhaustive.

Yep, [here's a demo](https://flow.org/try/#0KYOwrgtgBAolDeUCCAaKAhKBfA3AKAApgAuWASigF4A+BPKKAZwHcBLAFwGMALKIi+PQZROAQ0bBYAOiSkARgCdgogNb5hUAPSaR4yTCnp5S1euHaoAE2AAzUWAA27Y8rVCseD0A):
```js
enum E { A, B };
(e: E) => {
  switch (e) {
    case E.A: break;
    // case E.B: break;
    // default: break;
  }
}
```

That gives an error at the `switch (e)`:
> Incomplete exhaustive check: the member `B` of enum `E` [1] has not been considered in check of `e`. [invalid-exhaustive-check]

If you uncomment either the other case, or the default, the error goes away because now it's exhaustive.

I believe what that flowlint does is just to say also you can't have a default case (without explicitly suppressing the lint rule): you have to name all the enum members, so that when someone adds another member to that enum they definitely get notified of each of the switches on it.